numpy/core/_internal.py

```diff
@@ -305,6 +305,106 @@ def _index_fields(ary, fields):
     copy_dtype = {'names':view_dtype['names'], 'formats':view_dtype['formats']}
     return array(view, dtype=copy_dtype, copy=True)
 
+# get the list of offsets of every field in a dtype
+def get_all_field_offests(dtype, base_offset=0):
+    fields = []
+    if dtype.fields is not None:
+        for name in dtype.names:
+           sub_dtype = dtype.fields[name][0]
+           sub_offset = dtype.fields[name][1] + base_offset
+           fields.extend(get_all_field_offests(sub_dtype, sub_offset))
+    else:
+        if dtype.shape:
+            sub_offsets = get_all_field_offests(dtype.base, base_offset)
+            count = 1
+            for dim in dtype.shape:
+                count *= dim
+            fields.extend((typ,off + dtype.base.itemsize*j)
+                                              for j in range(count)
+                                              for typ,off in sub_offsets)
+        else:
+            fields.append((dtype, base_offset))
+    return fields
+
+# perform object memory overlap tests (see _view_is_safe)
+def check_field_overlap(new_fields, old_fields):
+    from .numerictypes import object_
+    from .multiarray import dtype
+
+    #first go byte by byte and check we do not access bytes not in old_fields
+    new_bytes = set()
+    for tp,off in new_fields:
+        new_bytes.update(set(range(off, off+tp.itemsize)))
+    old_bytes = set()
+    for tp,off in old_fields:
+        old_bytes.update(set(range(off, off+tp.itemsize)))
+    if len(new_bytes.difference(old_bytes)) != 0:
+        raise TypeError("view would access data parent array doesn't own")
+
+    #next check that we do not interpret non-Objects as Objects, and vv
+    obj_offsets = [off for tp,off in old_fields if tp == object_]
```

Comment:
Isn't the correct check here `if tp is object_`? I'm just raising the question, not really sure.